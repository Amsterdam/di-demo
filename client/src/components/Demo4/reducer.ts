const reducer = (state, action) => {
  console.log('reducer', state, action);

  switch (action.type) {
    case 'validateForm':
      return {
        ...state,
        formValid: true,
        owner: action.payload.owner
      };

    case 'invalidateForm':
      return {
        ...state,
        formValid: false,
      };

    case 'setProperties':
      return {
        ...state,
        hasResult: true,
        name: action.payload.name,
        street: action.payload.street,
        city: action.payload.city,
        telephone: action.payload.telephone,
      };

    default:
      return state;
  }
};

export default reducer;