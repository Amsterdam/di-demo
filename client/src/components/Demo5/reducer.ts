interface IState {
    hasResult?: boolean;
    hasError?: boolean;
    formErrors?: string[];
    location?: string;
    report?: string;
    phoneConsent?: boolean;
    updates?: boolean;
    phone?: string;
    email?: string;
}

interface IAction {
    type: string;
    payload?: IState;
}

export const initialState: IState = {
    hasResult: false,
    hasError: false,
    formErrors: [],
    location: '',
    report: '',
    phoneConsent: null,
    updates: null
};

export const reducer = (state: IState, action: IAction): IState => {
    switch (action.type) {
        case 'validateForm':
            return {
                ...state,
                location: action.payload.location,
                report: action.payload.report,
                phoneConsent: action.payload.phoneConsent,
                updates: action.payload.updates,
                formErrors: action.payload.formErrors
            };
        case 'setResult':
            return {
                ...state,
                hasResult: true,
                hasError: false,
                phone: action.payload.phone,
                email: action.payload.email
            };
        case 'setError':
            return {
                ...state,
                hasError: true
            };
        default:
            return state;
    }
};
