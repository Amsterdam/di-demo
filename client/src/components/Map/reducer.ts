import { LatLngLiteral, Map } from 'leaflet';

interface IState {
    mapInstance?: Map | null;
    url?: string;
    query?: string;
    autosuggest?: ILocation[] | null;
    latLng?: LatLngLiteral | null;
    location?: ILocation | null;
    showAutosuggest?: boolean;
}

export interface ILocation {
    id?: string;
    displayName: string;
    latLng?: LatLngLiteral;
}

interface IAction {
    type: string;
    payload?: {
        mapInstance?: Map | null;
        url?: string;
        query?: string;
        autosuggest?: ILocation[] | null;
        location?: ILocation | null;
        showAutosuggest?: boolean;
    };
}

export const initialState: IState = {
    mapInstance: null,
    url: '',
    query: '',
    autosuggest: null,
    location: null,
    showAutosuggest: false
};

export const reducer = (state: IState = initialState, action: IAction): IState => {
    switch (action.type) {
        case 'setAutosuggest':
            return {
                ...state,
                autosuggest: (action.payload as any).autosuggest,
                showAutosuggest: true
            };

        case 'setLocation':
            return {
                ...state,
                location: (action.payload as any).location
            };

        case 'hideAutosuggest':
            return {
                ...state,
                showAutosuggest: false
            };

        case 'setMapInstance':
            return {
                ...state,
                mapInstance: (action.payload as any).mapInstance
            };

        case 'onChangeLocation':
            return {
                ...state,
                query: (action.payload as any).query,
                url: (action.payload as any).url
            };
        default:
            return state;
    }
};
