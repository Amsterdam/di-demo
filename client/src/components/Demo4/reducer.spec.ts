import { initialState, reducer } from './reducer';

describe('Demo4 reducer', () => {
    it('should update when action validateForm is dispatched', () => {
        const newState = reducer(initialState, { type: 'validateForm', payload: { owner: 'test' } });
        expect(newState).toMatchSnapshot();
    });

    it('should update when action invalidateForm is dispatched', () => {
        const newState = reducer(initialState, { type: 'invalidateForm' });
        expect(newState).toMatchSnapshot();
    });

    it('should update when action setProperties is dispatched', () => {
        const newState = reducer(initialState, {
            type: 'setProperties',
            payload: { name: 'Test test', street: 'Teststraat', city: 'Amsterdam', telephone: '0612345678' }
        });
        expect(newState).toMatchSnapshot();
    });

    it('should update when action setError is dispatched', () => {
        const newState = reducer(initialState, { type: 'setError' });
        expect(newState).toMatchSnapshot();
    });

    it('should not update state when an unknown action is dispatched', () => {
        const newState = reducer(initialState, { type: 'unknown' });
        expect(newState).toMatchSnapshot();
    });
});
