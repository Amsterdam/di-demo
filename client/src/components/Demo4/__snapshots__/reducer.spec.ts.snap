// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo4 reducer should not update state when an unknown action is dispatched 1`] = `
Object {
  "city": "",
  "email": "",
  "formValid": true,
  "hasError": false,
  "hasResult": false,
  "name": "",
  "owner": "",
  "street": "",
  "telephone": "",
}
`;

exports[`Demo4 reducer should update when action invalidateForm is dispatched 1`] = `
Object {
  "city": "",
  "email": "",
  "formValid": false,
  "hasError": false,
  "hasResult": false,
  "name": "",
  "owner": "",
  "street": "",
  "telephone": "",
}
`;

exports[`Demo4 reducer should update when action setError is dispatched 1`] = `
Object {
  "city": "",
  "email": "",
  "formValid": true,
  "hasError": true,
  "hasResult": false,
  "name": "",
  "owner": "",
  "street": "",
  "telephone": "",
}
`;

exports[`Demo4 reducer should update when action setProperties is dispatched 1`] = `
Object {
  "city": "Amsterdam",
  "email": undefined,
  "formValid": true,
  "hasError": false,
  "hasResult": true,
  "name": "Test test",
  "owner": "",
  "street": "Teststraat",
  "telephone": "0612345678",
}
`;

exports[`Demo4 reducer should update when action validateForm is dispatched 1`] = `
Object {
  "city": "",
  "email": "",
  "formValid": true,
  "hasError": false,
  "hasResult": false,
  "name": "",
  "owner": "test",
  "street": "",
  "telephone": "",
}
`;
